client/cli/src/commands/generate.rs

```diff
@@ -0,0 +1,105 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2020 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Implementation of the `generate` subcommand
+use bip39::{MnemonicType, Mnemonic, Language};
+use structopt::StructOpt;
+use crate::{
+	print_from_uri, CliConfiguration, KeystoreParams, SharedParams, Error,
+	with_crypto_scheme, NetworkSchemeFlag, OutputTypeFlag, CryptoSchemeFlag,
+};
+
+/// The `generate` command
+#[derive(Debug, StructOpt)]
+#[structopt(name = "generate", about = "Generate a random account")]
+pub struct GenerateCmd {
+	/// The number of words in the phrase to generate. One of 12 default), 15, 18, 21 and 24.
+	#[structopt(long, short = "w", value_name = "WORDS")]
+	words: Option<usize>,
+
+	#[allow(missing_docs)]
+	#[structopt(flatten)]
+	pub keystore_params: KeystoreParams,
+
+	#[allow(missing_docs)]
+	#[structopt(flatten)]
+	pub shared_params: SharedParams,
```

Comment:
in order to impl `CliConfiguration` for it

Reply:
shared params is a requirement for it

Reply:
You can not just mark this as solved because `CliConfiguration` requires it! I know that `CliConfiguration` requires it. But this is a really shitty CLI experience. If you don't need this, you need to adapt `CliConfiguration`.